text/0000-prepare-operation.md

```diff
@@ -0,0 +1,387 @@
+
+ - **Remove `builder` from `io.buildpacks` namespace**
+
+[pd-02]: https://github.com/buildpacks/spec/blob/extensions/project-descriptor/0.2/extensions/project-descriptor.md#iobuildpacks-optional
+
+### Version `io.buildpacks` namespace
+
+The `io.buildpacks` namespace schema should be able to change without changing the Project Descriptor spec. For this reason, the namespace should define a `schema-version`.
+
+### Schema
+
+See [`io.buildpacks` Namespace Schema](#io-buildpacks-Namespace-Schema) section below.
+
+#### Usages Examples
+
+###### Multi-Platform (pack + kpack)
+
+In the following example we show how various platform namespaces can have their own configuration but properties can be promoted to `io.buildpacks` if the user's intent is to apply it to all platforms.
+
+```toml
+[io.buildpacks]
+schema-version = "0.2"
+
+###
+# common buildpacks config
+##
+
+[io.buildpacks]
+exclude = ["some-files/**"]
+
+##
+# pack config
+##
+
+[io.buildpacks.pack]
+schema-version = "0.1"
+builder = "cnbs/sample-builder:bionic"
+cache-volume = "my-app-cache"           # (example only)
+
+##
+# kpack config
+##
+
+[com.vmware.kpack]
+schema-version = "0.1"
+
+[[com.vmware.kpack.build.env]]
+name="CUSTOM_ENV"
+value="SOME_VALUE"
+```
+
+Notice:
+
+1. Both `pack` and `kpack` are expected to apply `io.buildpacks.exclude`.
+2. `io.buildpacks.pack` has additional properties that are specific to `pack` only.
+
+## Prepare phase
+
+A `prepare` phase would be a new lifecycle phase that gets executed before `create` or `detect` phase. Similar to the `rebaser`, it is outside of the standard [Build operation][build-operation].
+
+[build-operation]: https://github.com/buildpacks/spec/blob/main/platform.md#build
+
+#### Responsibilities
+
+At minimum, the expected reponsibility of the `preparer` would be to:
+
+ - Apply the requested configuration.
+
+#### Inputs
+
+The `prepare` phase would take all the same inputs as `creator` plus the addition of a path to `project.toml`.
+
+#### Outputs
+
+The `prepare` phase may affect the file system, and mutation or create files that would be consumed be futher phases.
+
+## Cloud Native Buildpacks utilities
+
+The Cloud Native Buildpacks project will provide the following functionality in the form of utilities:
+
+###### `go` prepare function
+
+A function that can help developers apply configuration to the filesystem based on a provided configuration. 
+
+> This would be useful for builder providers or platform implementers that choose to provide their own [`preparer` executable](#preparer-executable).
+
+###### `preparer` executable
+
+The Cloud Native Buildpacks project will have a `preparer` it ships along with the existing `lifecyle` image.
+
+> This would be useful for builder providers or platform implementers to not have to develop standard functionality.
+
+The default implementation COULD take care of applying the following configuration:
```

Comment:
To link up some discussion re: [https://github.com/buildpacks/spec/pull/288](https://github.com/buildpacks/spec/pull/288#issuecomment-1057283717), it could also download a run-image sbom if provided as a label pointing to a layer or a cosign-attached image manifest.